src/server.rs

```diff
@@ -935,9 +1006,9 @@ impl Server {
         );
     }
 
-    fn deserialize_log_entries(&self, data: &[u8]) -> LogEntry {
+    fn deserialize_log_entries(&self, data: &[u8]) -> Result<LogEntry, Box<ErrorKind>> {
```

Comment:
Here's what I'm thinking of doing.
```rust
    fn deserialize_log_entries(&self, data: &[u8]) -> Result<LogEntry> {
        bincode::deserialize(data).map_err(|e| Error::BincodeError(e))
    }
```

Reply:
Thanks @dierbei for the feedback.

I'll make the changes and update the PR today 👍

Reply:
Hi @dierbei,
I have incorporated the suggested changes in my latest PR #38.

In summary, I made the following changes:

1. Introduce a new `Error` variant called `Bincode` to wrap `bincode::Error`.
2. Modify the `deserialize_log_entries` function to use the new `Error` variant and incorporate your suggested change.

I have also handled the recent merge conflict.

Thanks

Reply:
Quick update on the changes:
1. I've removed my `Error` variant called `Bincode`, and I'm instead using the existing `BincodeError`, which performs the same function.
2. I've locally tested my changes with the commands in the `ci.yml` file, and everything works fine with no warnings or errors.

With this, the changes suggested in @dierbei's comment and the merge conflicts caused by recent changes in `main` have been resolved.

Hence, I am marking this comment as resolved.

Thanks.